src/vaults.lua

```diff
@@ -1,28 +1,30 @@
-Vaults = Vaults or {}
-
 -- Utility functions that modify global Vaults object
 local vaults = {}
 local balances = require("balances")
 local utils = require("utils")
 local constants = require("constants")
 
---- @type Timestamp|nil
-NextBalanceVaultsPruneTimestamp = NextBalanceVaultsPruneTimestamp or 0
-
 --- @class Vault
---- @field balance number The balance of the vault
---- @field startTimestamp number The start timestamp of the vault
---- @field endTimestamp number The end timestamp of the vault
+--- @field balance mARIO The balance of the vault
+--- @field controller WalletAddress | nil The controller of a revokable vault. Nil if not revokable (default: nil)
+--- @field startTimestamp Timestamp The start timestamp of the vault
+--- @field endTimestamp Timestamp The end timestamp of the vault
+
+--- @alias Vaults table<WalletAddress, table<VaultId, Vault>> -- A table of vaults indexed by owner address, then by vault id
+
+--- @type Vaults
+Vaults = Vaults or {}
 
---- @class Vaults: table<string, Vault> A table of vaults indexed by owner address
```

Comment:
Ah, was this improperly typed before?

Reply:
yes, there were a couple type issues in the IDE that were resolved from this